cdn/signed-urls/src/main/java/com/google/cdn/SignedCookies.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cdn;
+
+// [START cloudcdn_sign_cookie]
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.security.InvalidKeyException;
+import java.security.Key;
+import java.security.NoSuchAlgorithmException;
+import java.util.Base64;
+import java.util.Calendar;
+import java.util.Date;
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+
+public class SignedCookies {
+
+  public static void main(String[] args) throws Exception {
+    // TODO(developer): Replace these variables before running the sample.
+
+    // The name of the signing key added to the back end bucket or service
+    String keyName = "YOUR-KEY-NAME";
+    // Path to the url signing key uploaded to the backend service/bucket, as a 16-byte array
+    String keyPath = "/path/to/key";
+    // The date that the signed URL expires
+    Date expirationTime = getTomorrow();
+    // URL of request
+    String requestUrl = "https://media.example.com/videos/id/main.m3u8?userID=abc123&starting_profile=1";
+    // URL prefix to sign as a string. urlPrefix must start with either http:// or
+    // https:// and should not include query parameters
+    String urlPrefix = "https://media.example.com/videos/";
```

Comment:
Consider renaming `urlPrefix` to be more meaningful.

Reply:
I am trying to be consistent with the documentation page this snippet shows up [here](https://cloud.google.com/cdn/docs/using-signed-cookies#urlprefix).